tests/test_table.py

```diff
@@ -110,38 +111,92 @@ def test_query(TestModel, TestModel_entries, dynamo_local):
+    results = list(TestModel.query(foo="first", bar__begins_with="t"))
     assert len(results) == 2
 
+    results = list(TestModel.query(query_kwargs={"Limit": 2}, foo="first"))
+    assert len(results) == 2
+    assert results[0].count == 111
+    assert results[1].count == 333
+
 
 def test_scan(TestModel, TestModel_entries, dynamo_local):
     """Scanning should return the expected values"""
-    results = TestModel.scan(count__gt=200)
+    results = list(TestModel.scan(count__gt=200))
     assert len(results) == 2
 
     # our table has a hash and range key, so our results are ordered based on the range key
     assert results[0].count == 333
     assert results[1].count == 222
 
-    results = TestModel.scan(child__sub="two")
+    results = list(TestModel.scan(child__sub="two"))
     assert len(results) == 1
     assert results[0].count == 222
 
-    results = TestModel.scan(child__sub__begins_with="t")
+    results = list(TestModel.scan(child__sub__begins_with="t"))
     assert len(results) == 2
     assert results[0].count == 333
     assert results[1].count == 222
 
+    results = list(TestModel.scan(scan_kwargs={"Limit": 2}, count__gt=0))
+    assert len(results) == 2
+    assert results[0].count == 111
+    assert results[1].count == 333
+
+    TestModel.put({"foo": "no_baz", "bar": "omg"})
+    results = list(TestModel.scan(baz__not_exists=True))
+    assert len(results) == 1
+    assert results[0].foo == "no_baz"
+
+    with pytest.raises(TypeError):
+        # Make sure we reject if the value isn't True
+        list(TestModel.scan(baz__not_exists=False))
+
+
+def test_yield_items(TestModel, mocker):
+    # Mock out Dynamo responses as each having only one item to test auto-paging
+    side_effects = [{
+        'Items': [{'bar': 'one', 'baz': 'bbq', 'child': {'sub': 'one'}, 'count': Decimal('111'), 'foo': 'first'}],
+        'LastEvaluatedKey': {'cookie_id': 'ec477c69-8bc5-4e14-995d-37a73e8eb185', 'created_at': Decimal('1490000000')},
+        'ScannedCount': 1
+        }, {
+        'Items': [{'bar': 'two', 'baz': 'bbq', 'child': {'sub': 'one'}, 'count': Decimal('222'), 'foo': 'second'}],
+        'ScannedCount': 1
+    }]
+    mocker.patch.object(TestModel.Table.__class__, 'scan', side_effect=side_effects)
+    results = list(TestModel._yield_items('scan', dynamo_kwargs={"Limit": 2}))
+
+    assert TestModel.Table.scan.call_count == 2
+    assert len(results) == 2
+    assert results[0].count == 111
+    assert results[1].count == 222
+
+    mocker.patch.object(TestModel.Table.__class__, 'query', side_effect=side_effects)
+    results = list(TestModel._yield_items('query', dynamo_kwargs={"Limit": 2}))
+
+    assert TestModel.Table.query.call_count == 2
+    assert len(results) == 2
+    assert results[0].count == 111
+    assert results[1].count == 222
+
+
+def test_yield_items_xlarge(TestModel, TestModel_entries_xlarge, dynamo_local, mocker):
+    mocker.spy(TestModel.Table.__class__, 'scan')
```

Comment:
This invocation of `spy` causes pypy to fail tests (i.e. https://travis-ci.org/NerdWalletOSS/dynamorm/jobs/229322568)

It's interesting that it passes on cpython, and shows some of the subtle implementation differences in pypy!  The problem is that `scan` is not a class method, so you really want to spy on the instance:

```
mocker.spy(TestModel.Table, 'scan')
```

(This is true on line 176 & 184 too, and I tested the changes locally on both pypy & cpython)

In cpython this still ends up spying on the instance created correctly (some how), but per the data model spec (https://docs.python.org/2/reference/datamodel.html#object.__getattribute__) pypy is [doing the correct thing and] failing because it didn't get an instance of a class.